src/components/CustomerQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FunctionComponent } from "react";
+import { Transaction } from "../../services/quota";
+import { differenceInSeconds, format, formatDistance } from "date-fns";
+import { View } from "react-native";
+import { CustomerCard } from "./CustomerCard";
+import { AppText } from "../Layout/AppText";
+import { color } from "../../common/styles";
+import { sharedStyles } from "./sharedStyles";
+import { DarkButton } from "../Layout/Buttons/DarkButton";
+
+const DURATION_THRESHOLD_SECONDS = 60 * 10; // 10 minutes
+
+interface NoQuotaCard {
+  nric: string;
+  remainingQuota: Transaction[];
+  onCancel: () => void;
+}
+
+/**
+ * Shows when the user cannot purchase anything
+ *
+ * Precondition: Only rendered when remainingQuota are all 0
+ */
+export const NoQuotaCard: FunctionComponent<NoQuotaCard> = ({
+  nric,
+  remainingQuota,
+  onCancel
+}) => {
+  const now = new Date();
+  const secondsFromLastTransaction = remainingQuota[0]?.transactionTime
+    ? differenceInSeconds(now, new Date(remainingQuota[0].transactionTime))
+    : -1;
+  return (
+    <View>
+      <CustomerCard nric={nric} headerBackgroundColor={color("red", 60)}>
+        <View
+          style={[
+            sharedStyles.resultWrapper,
+            sharedStyles.failureResultWrapper
+          ]}
+        >
+          <AppText style={sharedStyles.emoji}>❌</AppText>
+          <AppText style={sharedStyles.statusTitleWrapper}>
+            {secondsFromLastTransaction > 0 ? (
+              secondsFromLastTransaction > DURATION_THRESHOLD_SECONDS ? (
+                <>
```

Comment:
Was thinking if you want to put this in a separate component with names like:

RecentTransactionPanel
DistantTransactionPanel
NoPreviousTransactionPanel

Reply:
Yep sure!